src/test/kotlin/com/terraformation/backend/tracking/db/plantingSiteStore/PlantingSiteStoreApplyEditTest.kt

```diff
@@ -0,0 +1,643 @@
+                  subzone { repeat(7) { cluster() } }
+                }
+              },
+          desired = newSite(width = 750),
+          expected =
+              newSite(width = 750) {
+                zone {
+                  extraPermanentClusters = 3
+                  numPermanentClusters = 10
+                  subzone()
+                }
+              },
+          expectedPlotCounts =
+              listOf(
+                  rectangle(x = 0, width = 500, height = 500) to 7 * 4,
+                  rectangle(x = 500, width = 250, height = 500) to 3 * 4,
+              ))
+    }
+
+    @Test
+    fun `uses random cluster numbers for newly added clusters, renumbering existing clusters if needed`() {
+      val existingClusterNumbers = mutableSetOf<Int>()
+      val newClusterNumbers = mutableSetOf<Int>()
+      val maxAttempts = 100
+      var currentAttempt = 0
+      val initialArea = rectangle(x = 0, width = 500, height = 500)
+      val addedArea = rectangle(x = 500, width = 500, height = 500)
+
+      while (++currentAttempt < maxAttempts &&
+          (newClusterNumbers != setOf(1, 2, 3, 4) || existingClusterNumbers != setOf(1, 2, 3, 4))) {
+        val (edited) =
+            runScenario(
+                initial =
+                    newSite(width = 500) {
+                      name = "Site $currentAttempt"
+                      zone {
+                        numPermanentClusters = 2
+                        subzone {
+                          cluster()
+                          cluster()
+                        }
+                      }
+                    },
+                desired = newSite(width = 1000),
+                expected =
+                    newSite(width = 1000) {
+                      name = "Site $currentAttempt"
+                      zone {
+                        extraPermanentClusters = 2
+                        numPermanentClusters = 4
+                        subzone()
+                      }
+                    },
+                expectedPlotCounts =
+                    listOf(
+                        initialArea to 8,
+                        addedArea to 8,
+                    ))
+
+        val monitoringPlots = edited.plantingZones[0].plantingSubzones[0].monitoringPlots
+
+        monitoringPlots.forEach { plot ->
+          plot.permanentCluster?.let { permanentCluster ->
+            if (plot.boundary.nearlyCoveredBy(addedArea)) {
+              newClusterNumbers.add(permanentCluster)
+            } else {
+              existingClusterNumbers.add(permanentCluster)
+            }
+          }
+        }
+
+        val plotCountsByClusterNumber =
+            monitoringPlots.groupBy { it.permanentCluster }.mapValues { (_, plots) -> plots.size }
+
+        assertEquals(
+            mapOf(1 to 4, 2 to 4, 3 to 4, 4 to 4),
+            plotCountsByClusterNumber,
+            "Number of plots with each cluster number after edit")
+      }
+
+      assertEquals(
```

Comment:
I'm not sure what the difference is between newClusterNumbers and existingClusterNumbers here.  We've expanded the area and added 2 new clusters.  What is this checking?

Reply:
Right, this test wasn't super clear. I added an explanation of what it's doing; please give that a once-over and see if it makes sense now.

Reply:
Yep, very clear now, thank you!